includes/class-newspack-popups-inserter.php

```diff
@@ -28,14 +28,22 @@ public static function popup_for_post() {
 		if ( self::$popup ) {
 			return self::$popup;
 		}
+
 		// First try for pop-up with category filtering.
-		self::$popup = Newspack_Popups_Model::retrieve_popup( get_the_category() );
+		$categories = get_the_category();
+		if ( $categories && count( $categories ) ) {
+			self::$popup = Newspack_Popups_Model::retrieve_popup( get_the_category() );
+			if ( self::$popup ) {
+				return self::$popup;
+			}
+		}
 
 		// If nothing found, try for sitewide pop-up.
-		if ( ! self::$popup ) {
-			self::$popup = Newspack_Popups_Model::retrieve_popup();
+		$sitewide_default = get_option( Newspack_Popups::NEWSPACK_POPUPS_SITEWIDE_DEFAULT, null );
+		if ( $sitewide_default ) {
+			self::$popup = Newspack_Popups_Model::retrieve_popup_by_id( $sitewide_default );
```

Comment:
Missing return statement:
```suggestion
			self::$popup = Newspack_Popups_Model::retrieve_popup_by_id( $sitewide_default );
			if ( self::$popup ) {
				return self::$popup;
			}
```

Reply:
Fixed in https://github.com/Automattic/newspack-popups/pull/50/commits/a4eb444c015aee7c61218384660fa82c5d5f2926